util/predicates/cluster_predicates.go

```diff
@@ -188,3 +188,27 @@ func ClusterUnpausedAndInfrastructureReady(logger logr.Logger) predicate.Funcs {
 	// Use any to ensure we process either create or update events we care about
 	return Any(log, createPredicates, updatePredicates)
 }
+
+// ClusterIsManagedTopology returns a Predicate that returns true when cluster.Spec.Topology
+// is NOT nil and false otherwise.
+func ClusterIsManagedTopology(logger logr.Logger) predicate.Funcs {
+	return predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
```

Comment:
Q: (for @vincepri @sbueringer ) I was expecting to apply the same func for Update/Create/Delete (And Generic) or I'm missing something at controller-runtime/client-go level?

Reply:
If all of them behave the same, use https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/predicate/predicate.go#L100-L118

Reply:
I think we can use `NewPredicateFuncs`

Reply:
Maybe it's better to follow the pattern in `ResourceNotPaused` so we don't lose the eventType information for the logger.

P.S. @fabriziopandini 
```
return processIfNotPaused(logger.WithValues("predicate", "updateEvent"), e.ObjectNew)
```
should be
```
return processIfNotPaused(logger.WithValues("predicate", "ResourceNotPaused", "eventType", "updateEvent"), e.ObjectNew)
```
right? (I can open a separate PR for it)

Reply:
@budhirajamadhav I would prefer using the pattern we use in `ResourceNotPaused` because this way we are able to log the correct event. That is not possible using `NewPredicateFuncs`